### PR TITLE
Use Bottom Sheet on Home Screen

### PR DESCRIPTION
**Pull request by wojtus7**
Use bottom sheet on home screen. Used library needs react-native-reanimated and react-native-gesture-handler so I needed to add them to the project. 

To make it work do `npm i` and rebuild native app. 

https://github.com/daimo-eth/daimo/assets/42337257/0357cc93-9230-486b-8210-3f24313d5990

### CODE REVIEW

File: apps/daimo-mobile/src/vendor/SwipeUpDown.tsx
Comment (nalinbhardwaj) [inline apps/daimo-mobile/src/vendor/SwipeUpDown.tsx:1]: nit, looks like this is no longer vendored so maybe we should move it to shared folder?

File: apps/daimo-mobile/src/vendor/SwipeUpDown.tsx [hunk 1]
```diff
@@ -1,55 +1,16 @@
...
-  Dimensions,
-  LayoutAnimation,
-  PanResponder,
-  PanResponderGestureState,
-  StyleSheet,
-  View,
-} from "react-native";
+import BottomSheet, { BottomSheetBackdrop } from "@gorhom/bottom-sheet";
+import { BottomSheetDefaultBackdropProps } from "@gorhom/bottom-sheet/lib/typescript/components/bottomSheetBackdrop/types";
+import { useBottomTabBarHeight } from "@react-navigation/bottom-tabs";
```
Comment (dcposch) [inline apps/daimo-mobile/src/vendor/SwipeUpDown.tsx:3]: 👍

**Review by dcposch — APPROVED** [review]
Looks great to me

Recommend moving from `vendor/` to `shared/`

When history is open, does tapping Home close it?

File: apps/daimo-mobile/src/view/shared/SwipeUpDown.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    const tabBarHeight = useBottomTabBarHeight();
+    const bottomRef = useRef<BottomSheet>(null);
+
+    const maxHeight = screenDimensions.height - ins.top - ins.bottom;
+    const posYMini = swipeHeight;
+    const posYFull = maxHeight - tabBarHeight;
+
+    const [isMini, setIsMini] = useState(true);
+
+    const animatedIndex = useSharedValue(0);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/shared/SwipeUpDown.tsx:49]: my first time seeing `useSharedValue` / `useImperativeHandle`, looks clean

File: apps/daimo-mobile/src/view/screen/HomeScreen.tsx [hunk 3]
```diff
@@ -30,6 +31,24 @@ export default function HomeScreen() {
 }
 
 function HomeScreenInner({ account }: { account: Account }) {
+  const bottomSheetRef = useRef<SwipeUpDownRef>(null);
+  const nav = useNav();
+  const isFocused = useIsFocused();
+
+  useEffect(() => {
+    if (nav.getParent()) {
+      // @ts-ignore
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HomeScreen.tsx:40]: all equal i recommend local fixes (eg `e: any`) vs @ts-ignore for the whole line

**Review by dcposch — APPROVED** [review]
LGTM

**Pull request merged**